components/dash-core-components/src/components/Geolocation.react.js

```diff
@@ -0,0 +1,193 @@
+import {Component} from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * The CurrentLocation component gets geolocation of the device from the web browser.  See more info here:
+ * https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API
+ */
+
+export default class Geolocation extends Component {
+    constructor(props) {
+        super(props);
+        this.success = this.success.bind(this);
+        this.error = this.error.bind(this);
+    }
+    updatePosition() {
+        if (this.props.update_now) {
+            this.props.setProps({
+                update_now: false,
+            });
+        }
+        if (!navigator.geolocation) {
+            this.error({
+                code: 999,
+                message: 'Your browser does not support Geolocation',
+            });
+            if (this.props.show_alert) {
+                alert(`ERROR(${this.error.code}): ${this.error.message}`);
+            }
```

Comment:
```suggestion
```
Already handled inside `this.error` - and I don't think the version here would work anyway?

Reply:
It works - it's just an option to show the error as an alert instead of being buried in the console.  If you don't think this is a good idea, I should remove the prop as well.